### ISSUE TITLE
Incremental build breaks because model loader tries to load lazy model instead of compiled model

### ISSUE THREAD

**Issue by FroMage** [ceylon/ceylon-compiler#1085]
Suppose we have type `A`, used in the API of `B`, which is in turn used in `C`. We compile them all at once and it works.

Then we compile only `A` and `C`. When `C` needs `B` we load it from the model loader, from its class. At some point we're seeing the type of `A` in its API, and javac is going to try to _complete_ its `ClassSymbol` by parsing it, which triggers a fail-safe exception that we put in place to detect this.

Basically, our model loader is broken when it should not load things from class because we're compiling them. This would take too long to fix properly for M5.
